helm/cluster-vsphere/templates/_helpers.tpl

```diff
@@ -34,3 +61,79 @@ Create a prefix for all resource names.
 {{- define "resource.default.name" -}}
 {{ .Values.cluster.name }}
 {{- end -}}
+
+{{- define "kubeletExtraArgs" -}}
+{{- .Files.Get "files/kubelet-args" -}}
+{{- end -}}
+
+{{/*
+Updates in KubeadmConfigTemplate will not trigger any rollout for worker nodes.
+It is necessary to create a new template with a new name to trigger an upgrade.
+See https://github.com/kubernetes-sigs/cluster-api/issues/4910
+See https://github.com/kubernetes-sigs/cluster-api/pull/5027/files
+*/}}
+{{- define "kubeadmConfigTemplateSpec" -}}
+  {{- with $.Values.ssh.users }}
+users:
+    {{- range . }}
+- name: {{ .name }}
+  sshAuthorizedKeys:
+      {{- range .authorizedKeys }}
+  - {{ . }}
+      {{- end }}
+  sudo: ALL=(ALL) NOPASSWD:ALL
+    {{- end -}}
+  {{- end }}
+joinConfiguration:
+  nodeRegistration:
+    criSocket: /run/containerd/containerd.sock
+    kubeletExtraArgs:
+      {{- include "kubeletExtraArgs" . | nindent  6}}
+      node-labels: "giantswarm.io/node-pool={{ .pool.name }}"
+files:
+  {{- if $.Values.proxy.enabled }}
+    {{- include "containerdProxyConfig" . | nindent 2}}
```

Comment:
Is the definition of `containerdProxyConfig` missing?

Reply:
done in commit https://github.com/giantswarm/cluster-vsphere/pull/23/commits/19db703f0804b951714f04d366e681bddc4f9573 (on this PR)